route/src/route_generator_worker.cpp

```diff
@@ -438,5 +438,51 @@ namespace route {
         return true;
     }
 
+    bool RouteGeneratorWorker::crosstrack_error_check(const geometry_msgs::PoseStampedConstPtr& msg, lanelet::ConstLanelet current)
+    {
+  
+       lanelet::BasicPoint2d position;
+
+        position.x()= msg->pose.position.x;
+        position.y()= msg->pose.position.y;
+
+        if(boost::geometry::within(position, current.polygon2d())) //If vehicle is inside current_lanelet, there is no crosstrack error
+        {
+            return false;
+        }
+
+        ROS_DEBUG_STREAM("LLt Polygon Dimensions1: " << current.polygon2d().front().x()<< ", "<< current.polygon2d().front().y());
+        ROS_DEBUG_STREAM("LLt Polygon Dimensions2: " << current.polygon2d().back().x()<< ", "<< current.polygon2d().back().y());
+        ROS_DEBUG_STREAM("Distance1: "<< boost::geometry::distance(position, current.polygon2d())<<" Crosstrack: "<< cross_track_dist );
+    
+        if (boost::geometry::distance(position, current.polygon2d()) > cross_track_dist) //Evaluate lanelet crosstrack distance from vehicle
+            {
+                cte_count_++;
+
+                if(cte_count_ > cte_count_max_) //If the distance exceeds the crosstrack distance a certain number of times, report that the route has been departed
+                    return true;
+            }
```

Comment:
let's add `else logic` here that resets the count with a similar reasoning in the comment above.
`else
{
cte_count_ = 0;
}
`